src/components/QRCodeModal/index.tsx

```diff
@@ -1,27 +1,45 @@
+"use client";
+
 import React from "react";
-import Image, { StaticImageData } from "next/image";
+
+import { UserWithProfile } from "@/types/UserWithProfile";
 
 import { useDisableBodyScroll } from "../../hooks/disableBackgroundMoving";
+import QRCodeTab from "../QRCodeTab";
+import PerfilTab from "../QRCodeTab/PerfilTab";
+import ScanTab from "../QRCodeTab/ScanTab";
 
-import { BsFillClipboardFill, BsQrCodeScan, BsX } from "react-icons/bs";
+import { BsX } from "react-icons/bs";
 
 interface QRCodeModalProps {
   hidden: boolean;
   setHidden: React.Dispatch<React.SetStateAction<boolean>>;
+  user: UserWithProfile;
 }
-const QRCodeModal: React.FC<QRCodeModalProps> = ({ hidden, setHidden }) => {
+const QRCodeModal: React.FC<QRCodeModalProps> = ({
+  hidden,
+  setHidden,
+  user,
+}) => {
+  // disable body scroll
   useDisableBodyScroll({ modalIsHidden: hidden });
 
-  // Prevent closing the modal when clicking on its content
-  const handleModalClick = (
-    event: React.MouseEvent<HTMLDivElement, MouseEvent>
-  ) => {
-    event.stopPropagation();
-  };
+  const tabTitles = ["Perfil", "Scan"];
+  const tabs = [<PerfilTab user={user} />, <ScanTab user={user} />];
+  const modalTitle = [
+    <>
+      Partilha o teu <span className="text-primary">Perfil</span>
```

Comment:
Maybe the best approach to both of these modal titles would be something like:

- Partilha o teu QRCODE
- Dá scan a um QRCODE


I think it sounds less weird, idk

Reply:
yeah I agree. tbh I was very undecided about what titles to choose, ty for the feedback!